example/monteCarloIntegration/src/monteCarloIntegration.cpp

```diff
@@ -0,0 +1,207 @@
+            i < numPoints;
+            i += globalThreadExtent.prod())
+        {
+            // Generate a point in the 2D interval.
+            float x = dist(generator);
+            float y = dist(generator);
+            // Count every time where the point is "below" the given function.
+            if(y <= functor(
+                acc,
+                x))
+            {
+                ++localCount;
+            }
+        }
+
+        // Add the local result to the sum of the other results.
+        alpaka::atomic::atomicOp<
+            alpaka::atomic::op::Add>(
+            acc,
+            globalCounter,
+            localCount,
+            alpaka::hierarchy::Blocks{});
+    }
+};
+
+
+auto main() -> int
+{
+    // Defines and setup.
+    using Dim = alpaka::dim::DimInt<1>;
+    using Idx = std::size_t;
+    using Vec = alpaka::vec::Vec<
+        Dim,
+        Idx>;
+    using Acc = alpaka::example::ExampleDefaultAcc<
+        Dim,
+        Idx>;
+    using Host = alpaka::dev::DevCpu;
+    auto const devAcc = alpaka::pltf::getDevByIdx<Acc>(0u);
+    auto const devHost = alpaka::pltf::getDevByIdx<Host>(0u);
+    using QueueProperty = alpaka::queue::Blocking;
+    using QueueAcc = alpaka::queue::Queue<
+        Acc,
+        QueueProperty>;
+    QueueAcc queue{devAcc};
+
+    using BufHost = alpaka::mem::buf::Buf<
+        Host,
+        uint32_t,
+        Dim,
+        Idx>;
+    using BufAcc = alpaka::mem::buf::Buf<
+        Acc,
+        uint32_t,
+        Dim,
+        Idx>;
+    using WorkDiv = alpaka::workdiv::WorkDivMembers<
+        Dim,
+        Idx>;
+    // Problem parameter.
+    constexpr size_t numPoints = 100000000u;
+    constexpr size_t extent = 1u;
+    constexpr size_t numThreads = 100u; // Kernel will decide numCalcPerThread.
+    constexpr size_t numAlpakaElementsPerThread = 1;
+    WorkDiv workdiv{
+        alpaka::workdiv::getValidWorkDiv<Acc>(
+            devAcc,
+            Vec(numThreads),
+            Vec(numAlpakaElementsPerThread),
+            false,
+            alpaka::workdiv::GridBlockExtentSubDivRestrictions::Unrestricted)};
+
+    // Setup buffer.
+    BufHost bufHost{
+        alpaka::mem::buf::alloc<
+            uint32_t,
+            Idx>(
+            devHost,
+            extent)};
+    uint32_t *const ptrBufHost{alpaka::mem::view::getPtrNative(bufHost)};
+    BufAcc bufAcc{
+        alpaka::mem::buf::alloc<
+            uint32_t,
+            Idx>(
+            devAcc,
+            extent)};
+    uint32_t *const ptrBufAcc{alpaka::mem::view::getPtrNative(bufAcc)};
+
+    Kernel kernel;
+    alpaka::kernel::exec<Acc>(
+        queue,
+        workdiv,
+        kernel,
+        numPoints,
+        ptrBufAcc,
+        Function{});
+    alpaka::mem::view::copy(
+        queue,
+        bufHost,
+        bufAcc,
+        extent);
+    alpaka::wait::wait(queue);
+
+    // Check the result.
+    uint32_t globalCount = *ptrBufHost;
+    std::cout << "global count: " << globalCount << "\n";
```

Comment:
I think this output could be removed. As for a user the latter output of exact and actual result is informative, but this one is not necessarily.